File: src/test/java/com/frightanic/smn/api/geojson/CrsTest.java
package com.frightanic.smn.api.geojson;

import org.junit.Test;

import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;

/**
 * Tests Crs.
 */
public class CrsTest {

  /**
   * See method name.
   */
  @Test
  public void shouldHaveTypeMemberWithValueName() {
    // given
    Crs crs = new Crs(CrsType.CH1903);
    // when
    // then
    assertThat(crs.getType(), is("name"));
  }

  /**
   * See method name.
   */
  @Test
  public void shouldPlaceCrsTypeUrnInPropertyWithKeyName() {
    // given
    Crs crs = new Crs(CrsType.CH1903);
    // when
    // then
    assertThat(crs.getProperties().get("name"), is(CrsType.CH1903.getUrn()));
  }
}

File: src/test/kotlin/com/frightanic/smn/api/geojson/CrsTest.kt
package com.frightanic.smn.api.geojson

import com.frightanic.smn.api.geojson.CrsType.CH1903
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.junit.jupiter.api.Test

/**
 * Tests Crs.
 */
internal class CrsTest {
    @Test
    fun `should have type member with value name`() {
        // given
        val crs = Crs(CH1903)
        // when
        // then
        assertThat(crs.type, `is`("name"))
    }

    @Test
    fun `should place CRS type URN in property with key name`() {
        // given
        val crs = Crs(CH1903)
        // when
        // then
        assertThat(crs.properties["name"], `is`(CH1903.urn))
    }
}
